internal/planner/consumer_test.go

```diff
@@ -0,0 +1,888 @@
+		},
+	}, {
+		name:    "discards non-self-hosted jobs",
+		setupDB: func(db *fakeDB) {},
+		delivery: amqp.Delivery{
+			Headers: amqp.Table{"X-GitHub-Event": "workflow_job"},
+			Body: mk(map[string]any{
+				"action": "queued",
+				"repository": map[string]any{
+					"full_name": "canonical/test",
+				},
+				"workflow_job": map[string]any{
+					"id":         25,
+					"labels":     []string{"ubuntu-latest"},
+					"status":     "queued",
+					"created_at": "2025-01-01T00:00:00Z",
+				},
+			}),
+		},
+		expectErr: false,
+		checkDB: func(t *testing.T, db *fakeDB) {
+			assert.Nil(t, db.jobs["github:25"], "non-self-hosted job should not be inserted")
+		},
+		checkMetrics: func(t *testing.T, mr *telemetry.TestMetricReader) {
+			m := mr.Collect(t)
+			// 1 error from nack failure
+			assert.Equal(t, 1.0, m.Counter(t, webhookErrorsMetricName))
+			assert.Equal(t, 0.0, m.Counter(t, consumedWebhooksMetricName))
+			assert.Equal(t, 1.0, m.Counter(t, discardedWebhooksMetricName))
+		},
+	}, {
+		name:    "ignores waiting action",
+		setupDB: func(db *fakeDB) {},
+		delivery: amqp.Delivery{
+			Headers: amqp.Table{"X-GitHub-Event": "workflow_job"},
+			Body: mk(map[string]any{
+				"action": "waiting",
+				"repository": map[string]any{
+					"full_name": "canonical/test",
+				},
+				"workflow_job": map[string]any{
+					"id":         26,
+					"labels":     []string{"self-hosted"},
+					"status":     "waiting",
+					"created_at": "2025-01-01T00:00:00Z",
+				},
+			}),
+		},
+		expectErr: false,
+		checkDB: func(t *testing.T, db *fakeDB) {
+			assert.Nil(t, db.jobs["github:26"], "waiting action should not be processed")
+		},
+		checkMetrics: func(t *testing.T, mr *telemetry.TestMetricReader) {
+			m := mr.Collect(t)
+			// 1 error from nack failure
+			assert.Equal(t, 1.0, m.Counter(t, webhookErrorsMetricName))
+			assert.Equal(t, 0.0, m.Counter(t, consumedWebhooksMetricName))
+			assert.Equal(t, 1.0, m.Counter(t, discardedWebhooksMetricName))
+		},
+	}}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mr := telemetry.AcquireTestMetricReader(t)
+			defer telemetry.ReleaseTestMetricReader(t)
+
+			db := newFakeDB()
+			tt.setupDB(db)
+
+			ch := make(chan amqp.Delivery, 1)
+			ch <- tt.delivery
+			close(ch)
+
+			fakeConsumer := &fakeAmqpConsumer{
+				deliveries: ch,
+				pullErr:    tt.pullErr,
+			}
+			consumer := &JobConsumer{
+				consumer: fakeConsumer,
+				db:       db,
+				metrics:  NewMetrics(&fakeStore{}),
+			}
+
+			ctx := context.Background()
+			err := consumer.pullMessage(ctx)
+
+			if tt.expectErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+			if tt.checkDB != nil {
+				tt.checkDB(t, db)
+			}
+			if tt.checkMetrics != nil {
+				tt.checkMetrics(t, mr)
+			}
+		})
+	}
+}
+
+func TestParseGithubWebhookPayload(t *testing.T) {
+	mk := func(m map[string]any) []byte { b, _ := json.Marshal(m); return b }
```

Comment:
https://github.com/canonical/github-runner-operators/blob/main/STYLE.md#comments-in-tests

Reply:
Updated, thanks!